bson/src/main/org/bson/RawBsonDocument.java

```diff
@@ -395,4 +399,35 @@ private Object readResolve() {
             return new RawBsonDocument(bytes);
         }
     }
+
+    private CodecRegistry createCodecRegistry() {
+        HashMap<Class<?>, Codec<?>> replacementsForDefaults = new HashMap<Class<?>, Codec<?>>();
+        replacementsForDefaults.put(BsonDocument.class, new RawBsonDocumentNoCopyCodec());
+        return fromProviders(new BsonValueCodecProvider(replacementsForDefaults));
+    }
+
+    class RawBsonDocumentNoCopyCodec implements Codec<RawBsonDocument> {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public RawBsonDocument decode(final BsonReader reader, final DecoderContext decoderContext) {
+            BsonBinaryReader bsonReader = (BsonBinaryReader) reader;
```

Comment:
We can avoid exposing ourselves to potential changes in the way BsonBinaryReader handles the BsonInput (e.g. has it already read the size of the document from the input at this point) by adding a new method to BsonBinaryReader; something like getCurrentDocumentSize().  And then use the existing skipValue method to skip over the document.